### PR TITLE
Initializing new Team Configuration File

### PR DESCRIPTION
**Pull request by chinmdas**
Initializing new Team Configuration File and remove the obsolete web page,

**What It Does**

- No more obsolete web page when user tries to create a new CICS Profile. Instead a new global team config file will get initialized.

**How to Test**
- If NO global zowe.config.json file exists: Creates a New.
- If zowe.config.json file exists, but NO CICS Profile is present: Gives option to Edit.
- If zowe.config.json file exists and CICS Profile is available: Gives option to Edit.

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Comment by chinmdas** [comment]
This PR fixes https://github.com/zowe/cics-for-zowe-client/issues/33.

**Review by AndrewTwydell — CHANGES_REQUESTED** [review]
I get this popup when clicking "add" with no config file. We should change that text to something more meaningful.

image

**Review by AndrewTwydell — COMMENTED** [review]
When clicking edit config file when there are no profiles in there, we get a horrible error
image

image

**Comment by AndrewTwydell** [comment]
When you click the Add button on other ZE views and no config file exists, you get this menu
image

Perhaps we should do the same to be consistent, rather than going straight into the Global or Project quickpick??

**Review by Joe-Winchester — APPROVED** [review]

**Comment by chinmdas** [comment]
> I get this popup when clicking "add" with no config file. We should change that text to something more meaningful.
> 
> image

Yes, Agreed..!!! What can be appropriate info message? Do you think this makes sense?
`Team Configuration file not found. Creating new Team Configuration file`

**Comment by chinmdas** [comment]
> When clicking edit config file when there are no profiles in there, we get a horrible error image
> 
> image

Nice Testing @AndrewTwydell ..!!
The logic of picking up `filepath` for `showQuickPick` is very tightly coupled with profiles availability in `zowe.config.json` file. Since, you have delete all the profiles entry, extraction of `filepath` broke.
I am going to fix this. Thanks..!!

**Comment by chinmdas** [comment]
> When you click the Add button on other ZE views and no config file exists, you get this menu image
> 
> Perhaps we should do the same to be consistent, rather than going straight into the Global or Project quickpick??

Agreed...!!! Will implement this...!!

**Comment by chinmdas** [comment]
> Hi @AndrewTwydell ,
> 
> Where did you get a .vsix from to do the tests for the screen shots you've shown above ? I did a code review which looked good but I'd like to do a test of the PR code. Could checkout the branch and launch from VS code launch.json but wondering if there is a .vsix build created for the PR ?

Thanks @Joe-Winchester .... I see all the valid points raised by @AndrewTwydell . I would implement his suggestions and seek your approval again..!!

**Comment by Joe-Winchester** [comment]
On the CICS profile I still see the pop-up action `Update Profile` that opens the Connection Details dialog.  
The connection details dialog should be removed so it's not accessible at all (maybe just yank all the code), and instead `Update Profile` should probably drop into the editor for the `zowe.config.json` file itself.
(Note, this testing was me checking out the branch and doing a debug launch so it's possible I have a false test)

image

**Comment by chinmdas** [comment]
> On the CICS profile I still see the pop-up action `Update Profile` that opens the Connection Details dialog. The connection details dialog should be removed so it's not accessible at all (maybe just yank all the code), and instead `Update Profile` should probably drop into the editor for the `zowe.config.json` file itself. (Note, this testing was me checking out the branch and doing a debug launch so it's possible I have a false test, or also possible this fix is outside the scope of the issue this PR is fixing, and maybe another issue should be raised for removing the deprecated page for profile editing alltogether.
> 
> image

Okay.... Thank you bringing out this scenario @Joe-Winchester . I will fix this as well.

**Review by AndrewTwydell — CHANGES_REQUESTED** [review]
Looking good @chinmdas, few items to consider 👍🏼

**Comment by AndrewTwydell** [comment]
With the interest of trying to align with ZE's behaviour, there are a couple of questions to answer... 

There are currently 2 clicks required to update a profile (update -> edit) when that's the only option. ZE have 2 because they have manage -> edit. What are we aiming for? If we are eventually going to imitate them, should we also change to give the same manage profile menu?

Secondly, we no longer get the option to create a project config, only global. Same as above, are we trying to be consistent? Probably! 

Note: we shouldn't have the delete profile option anymore but that can be another PR to remove it 🙂

**Comment by chinmdas** [comment]
> With the interest of trying to align with ZE's behaviour, there are a couple of questions to answer...
> 
> There are currently 2 clicks required to update a profile (update -> edit) when that's the only option. ZE have 2 because they have manage -> edit. What are we aiming for? If we are eventually going to imitate them, should we also change to give the same manage profile menu?
> 
> Secondly, we no longer get the option to create a project config, only global. Same as above, are we trying to be consistent? Probably!
> 
> Note: we shouldn't have the delete profile option anymore but that can be another PR to remove it 🙂

Hi @Joe-Winchester @zFernand0 @davenice ,
Can we have your expert advice on this?
Just to build context, following is what this PR handles :

- `zowe.config.json` is missing -> `New Team Config File is created `-> _New Global Team Configuration file is created_
- `zowe.config.json` file is available, but CICS Profile is missing ->` Edit Team Config File` -> _Available Global Team Configuration file is opened for modification_
- `zowe.config.json` file is available + CICS Profile is available -> Edit Team Config File ->  _Available Global Team Configuration file is opened for modification_
- `Right click on CICS profile` -> `Update Profile` ->  _Available Global Team Configuration file is opened for modification_

So, The gaps that @AndrewTwydell has mentioned, should we do it as part of this PR or should we raise a new PR for that?

File: packages/vsce/src/trees/CICSTree.ts
Comment (davenice) [inline packages/vsce/src/trees/CICSTree.ts:119]: did you notice if there was a constant in Zowe Explorer we could use for zowe.confgi.json? or perhaps even for the whole profile path? that would be nice.

Comment (davenice) [reply packages/vsce/src/trees/CICSTree.ts:119]: Is this something we can use? Or not the right thing?
https://github.com/zowe/zowe-explorer-vscode/blob/aea6d92717f4b765088a8d17e36e0c2c1a38a8d4/packages/zowe-explorer/src/extending/ZoweExplorerExtender.ts#L99

**Review by davenice — COMMENTED** [review]
Looks good and seems good testing your branch out.

One question around whether we can use the Zowe Explorer code to get the path/filename 🤔

File: packages/vsce/src/trees/CICSTree.ts
Comment (davenice) [reply packages/vsce/src/trees/CICSTree.ts:119]: Actually I reckon that is not that useful, it needs us to give the basic path and it just adds the filename. Probably not worth using.

File: packages/vsce/src/trees/CICSTree.ts
Comment (AndrewTwydell) [inline packages/vsce/src/trees/CICSTree.ts:121]: It looks like filePath could be set to ""... Have you tested what happens when you call openConfigPath with an empty string? Might be worth reworking that so if can only ever be called with a valid parameter.

Comment (chinmdas) [reply packages/vsce/src/trees/CICSTree.ts:121]: Resolved..!!!

File: packages/vsce/src/trees/CICSTree.ts
Comment (chinmdas) [reply packages/vsce/src/trees/CICSTree.ts:119]: Resolving this thread..!!

File: packages/vsce/src/trees/CICSTree.ts
Comment (AndrewTwydell) [inline packages/vsce/src/trees/CICSTree.ts:574]: Sorry to be picky but, again, if osLoc is not set (which it doesn't have to be, it can be undefined), filePath can be set to undefined. Could do something similar to this, although you'd need to test and double check it, to only call openConfigFile with a valid input.

```suggestion
    if (response && profile.profLoc.osLoc?.[0]) {
      await openConfigFile(profile.profLoc.osLoc.[0]);
    } else {
      window.showErrorMessage("Some reasonable message about what you're trying to do??");
    }
```

File: packages/vsce/src/trees/CICSTree.ts
Comment (davenice) [inline packages/vsce/src/trees/CICSTree.ts:120]: just been discussing this with @chinmdas - we've decided that if the profile is not null, we should have profLoc and an entry in osLoc, therefore if we don't that's a problem and we should just fail here rather than in openConfigFile.

so we decided to remove the question mark.

File: packages/vsce/src/trees/CICSTree.ts
Comment (chinmdas) [reply packages/vsce/src/trees/CICSTree.ts:574]: Actually, that condition you have added is redundant.
Update profile scenario will only be invoked on a available profile. So, If profile is available >> the Team Configuration file is available >> no need to check on` profile.profLoc.osLoc?.[0]`

**Comment by AndrewTwydell** [comment]
The behaviour and flow of the profile experience is way better now @chinmdas, good work, I'm able to get to the config file when I need to and it handles odd edge cases nicely. I think it's a case of adding in those null checks that we've spoken about and the issue is resolved. We can create new issues for the problems we've found along the way 😉

Oh and are you able to squash your commits? Just to keep things as tidy as we can 😎

File: packages/vsce/src/trees/CICSTree.ts
Comment (chinmdas) [reply packages/vsce/src/trees/CICSTree.ts]: I have resolved this. We are falling back to Team Config File Path if osLoc fails to give us profile location path. This will cover us from all the angles.

File: packages/vsce/src/trees/CICSTree.ts
Comment (chinmdas) [reply packages/vsce/src/trees/CICSTree.ts]: I have resolved this. We are falling back to Team Config File Path if `osLoc` fails to give us profile location path. This will cover us from all the angles.

File: packages/vsce/src/trees/CICSTree.ts
Comment (AndrewTwydell) [inline packages/vsce/src/trees/CICSTree.ts:564]: A profile is always provided to this method right? Therefore we don't need an optional parameter here.

```suggestion
  async updateSessionHelper(profile: imperative.IProfAttrs) {
```

**Comment by davenice** [comment]
I think this looks pretty good - can you squash the commits down to 1 because it's just one piece of work?

That'll fix the DCO issue and make sure that whenever it gets merged it looks like one seamless piece? :-D

**Comment by chinmdas** [comment]
> enever it gets merged it looks like one seamless piece

I dont know, if I get 2 approvals, I can check if I have an option of squash merge.

**Comment by zFernand0** [comment]
no need to squash commit now 🙏 

You should be able to follow the preferred approach here:
- https://github.com/zowe/cics-for-zowe-client/pull/166/checks?check_run_id=34609378367

image

**Comment by AndrewTwydell** [comment]
@zFernand0 do you have squash and merge enabled on the repo, it would be nice to keep the history as tidy as possible right? 🫣 Avoiding a load of main merges and revert commits is preferable?

**Comment by Joe-Winchester** [comment]
> @zFernand0 do you have squash and merge enabled on the repo, it would be nice to keep the history as tidy as possible right? 🫣 Avoiding a load of main merges and revert commits is preferable?

Just checked the settings and it looks like it's enabled

image

**Review by davenice — APPROVED** [review]
Excellent. My preference is to organise commits before they're merged, simply because it's too easy for the person merging to click the wrong button and then my stream of commit-consciousness is listed in the history forever. Sadly I've been on both sides of that storyline!

But it sounds like we've got that covered today with the button :-) 👍

**Comment by chinmdas** [comment]
Hi @Joe-Winchester @zFernand0 , Do I have merge access?

**Review by AndrewTwydell — APPROVED** [review]
LGTM - Thanks @chinmdas! 

I don't think we have the powers to merge to main, no - potentially for the best for the time being!

**Comment by JillieBeanSim** [comment]
Hey @Joe-Winchester can add admin access to the team members you would like to be able to merge PRs 😸 
Should be able to do this under the Settings tab then setup group/users within the Access>Collaborators & Teams

**Comment by zFernand0** [comment]
As it stands right now, only these two teams have access to merge PRs to protected branches on this monorepo
- https://github.com/orgs/zowe/teams/zowe-cics-explorer-administrators (Maintainer access)
- https://github.com/orgs/zowe/teams/zowe-cli-administrators (Admin access)

As for adding more people to the mainteners/admins group(s), that's always up to the squad (it doesn't matter how new someone might be) 😋 

If y'all want to add a few more folks to the CICS Administrators, feel free to bring it up on the next standup 😋 

---

As for squashing vs merging...
I'm personally a fan of squashing PRs, however We may need to so some testing with the deployment workflow before we decide to move to a squash-approach 😋 

Thanks Joe for looking into the squash vs merge configuration.

**Comment by zFernand0** [comment]
Hey @chinmdas,
Did this commit resolve the problem DCO problem? c01d2b68d53d6b00ac36aa1ec13aabdbaf29072d 🥳 
Or did someone manually clicked the `Set DCO to Pass`? 🤔

**Review by zFernand0 — APPROVED** [review]
I really like the new flow 🙏 

As a follow up PR, we could have a Create CICS profile which just run the `zowe.all.config.init`with a warning to make people aware that new profiles will be created in their existing zowe.config.json file 😋 

(Similar to the Zowe Explorer flow) 😋

**Comment by zFernand0** [comment]
Labeling: `release-minor` to include these enhancements from the SDK
image

**Comment by chinmdas** [comment]
> Hey @chinmdas, Did this commit resolve the problem DCO problem? [c01d2b6](https://github.com/zowe/cics-for-zowe-client/commit/c01d2b68d53d6b00ac36aa1ec13aabdbaf29072d) 🥳 Or did someone manually clicked the `Set DCO to Pass` button? 🤔

Hi @zFernand0 ,
That commit did not pass DCO. I manually clicked `Set DCO to Pass` to check what option that gives me in further steps, but little I knew that one click will set the DCO to pass.

**Comment by chinmdas** [comment]
> I really like the new flow 🙏
> 
> As a follow up PR, we could have a Create CICS profile which just run the `zowe.all.config.init`with a warning to make people aware that new profiles will be created in their existing zowe.config.json file 😋
> 
> (Similar to the Zowe Explorer flow) 😋

Sure, We will do this.
Along with that, there were other scenario's that @AndrewTwydell and @davenice spotted which needs improvisation in order to stay in sync with ZE Flow. I will be creating new change set for that.
Thanks for your promptness @zFernand0 ..!!

**Comment by zFernand0** [comment]
> > Hey `@chinmdas`, Did this commit resolve the problem DCO problem? [c01d2b6](https://github.com/zowe/cics-for-zowe-client/commit/c01d2b68d53d6b00ac36aa1ec13aabdbaf29072d) 🥳 Or did someone manually clicked the `Set DCO to Pass` button? 🤔
> 
> Hi `@zFernand0` , That commit did not pass DCO. I manually clicked `Set DCO to Pass` to check what option that gives me in further steps, but little I knew that one click will set the DCO to pass.

Ideally, we should avoid clicking that button. There is just no configuration to remove that button. More details:
- https://github.com/dcoapp/app/issues/119

**Pull request merged**
